python/ray/services.py

```diff
@@ -580,6 +592,35 @@ def start_reaper():
     Returns:
         ProcessInfo for the process that was started.
     """
+    if sys.platform.startswith("linux"):
+        # No need for a reaper on Linux.
+        # The kernel supports fate-sharing natively, so just initilize it.
+        global libc
```

Comment:
Can we just do all of the checks at the top of this block and short circuit if they fail instead of having the global variables and checking if they're set?

Reply:
Not sure I understand what you mean. These `try`/`except`s are dependent on each other and need to happen sequentially. Could you share a snippet (or just push) the change you're thinking of?